crates/primitives/src/chain/mod.rs

```diff
@@ -151,6 +147,18 @@ impl Chain {
         Chain::Named(NamedChain::Dev)
     }
 
+    /// Returns true if the chain is contains Optimism configuration.
+    pub fn is_optimism(self) -> bool {
+        matches!(
+            self,
+            Chain::Named(NamedChain::Optimism) |
+                Chain::Named(NamedChain::OptimismGoerli) |
+                Chain::Named(NamedChain::OptimismKovan) |
+                Chain::Named(NamedChain::Base) |
+                Chain::Named(NamedChain::BaseGoerli)
+        )
+    }
```

Comment:
Don't really like this, feel like having the `optimism` bool flag was cleaner, wdyt @refcell?

Reply:
I also preferred the `optimism` bool flag here, felt more simple. Allows for easy additions of OP devnet chain configs etc.

Reply:
Agreed, prefer to keep it simple - we can push back on this :)

Reply:
Did matt explicitly ask us to remove the `optimism` bool?

Reply:
related comment: https://github.com/paradigmxyz/reth/pull/4377#discussion_r1376103265

Reply:
[Replied](https://github.com/paradigmxyz/reth/pull/4377#discussion_r1376936233) in the thread asking to keep the `optimism` bool flag.